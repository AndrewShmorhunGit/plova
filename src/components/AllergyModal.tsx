import styled from "styled-components";
import closeIcon from "../images/common/closeIcon.svg";

export const AllergyModal = ({
  setAllergyModal,
}: {
  setAllergyModal: React.Dispatch<React.SetStateAction<boolean>>;
}) => {
  const changeHandler = () => {};

  return (
    <Wrapper>
      <main className="modal-container show-modal">
        <div className="content center">
          <button className="close-btn" onClick={() => setAllergyModal(false)}>
            <img src={closeIcon} alt="close X" />
          </button>

          <h1>Your order</h1>
          <p>Add your allergies</p>

          <form className="form" action="#" onChange={changeHandler}>
            <textarea
              className="text-content"
              name="Allergy area"
              value=""
              placeholder="Let the restaurant know what they should take into account"
              maxLength={255}
            />
            <p className="characters-left">{255}</p>
          </form>

          <button className="btn center" onClick={() => setAllergyModal(false)}>
            Save
          </button>
        </div>
      </main>
    </Wrapper>
  );
};

const Wrapper = styled.main`
  .modal-container {
    position: fixed;
    inset: 0;
    background: rgba(0, 0, 0, 0.7);
    display: flex;
    justify-content: center;
    padding: 0rem 0 0 2rem;
    z-index: -1;
    overflow: auto;
  }

  .show-modal {
    z-index: 99;
    opacity: 1;
  }

  .content {
    background: #fff;
    position: relative;
    min-width: 60rem;
    max-height: 45rem;
    border-radius: 1rem;
    flex-direction: column;
    gap: 2rem;
    margin: 10vh auto;
    padding: 0;

    h1 {
      font-size: 3.2rem;
    }

    p {
      font-weight: 400;
    }

  }

  .form {
    position: relative;
    
  }

  .text-content {
    display: flex;
    background-color: #f5f5f5;
    width: 54rem;
    height: 18rem;
    border-radius: 2rem;
    padding 1rem;
    color: darkgrey;
    border: none;
    resize: none;
  }

  .characters-left {
    color: grey;
    position: absolute;
    bottom: 0.5rem;
    right: 1rem;
  }

  .btn {
    padding: 2.6rem 14rem;
    border-radius: 10rem;
  }

  .close-btn {
    position: absolute;
    top: 1.5rem;
    right: 1.5rem;
    background: transparent;
    border-color: transparent;
    font-size: 2rem;
    color: grey;
    cursor: pointer;
  }
`;
