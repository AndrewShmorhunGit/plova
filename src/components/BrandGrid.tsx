import styled from "styled-components";
import ratingExcellent from "../images/common/ratingExcellent.png";
import magnifyingGlass from "../images/design/magnifyingGlass.svg";
import storeDelivery from "../images/common/storeDelivery.svg";
import clocks from "../images/common/clocks.avif";
import { IMenu } from "../modules/modules";
import { showDollarPrice } from "../units/functions";
import { Cart, Products, Sections } from "./index";

export const BrandGrid: React.FC<{
  menu: IMenu;
  loading: boolean;
  error: string;
  selectedCategory: string | null;
}> = ({ menu, selectedCategory }) => {
  const getCurrentSubMenu = (categoryName: string | undefined) => {
    const currentMenu = menu.menu.find(
      (item) => item.category.categoryName === categoryName
    );
    if (currentMenu === undefined) {
      throw new Error("Sorry, something went wrong, currentMenu is undefined");
    }
    return currentMenu.products;
  };

  return (
    <Wrapper>
      <div className="container-products">
        <div className="grid-body">
          <div className="grid-title">
            <div className="brand-title">
              <h1>{menu.brandName}</h1>
              <div className="icons">
                <div className="del-icon-container center">
                  <img className="del-icon" src={clocks} alt="" />

                  <p className="icons-p">{`${menu.deliveryTime1}-${menu.deliveryTime2}'`}</p>
                </div>
                <div className="del-icon-container center">
                  <img
                    className="del-icon"
                    src={storeDelivery}
                    alt="delivery moped"
                  />
                  <p className="icons-p">
                    {showDollarPrice(menu.deliveryPrice)} $
                  </p>
                </div>
                <div className="del-icon-container center">
                  <img className="del-icon" src={ratingExcellent} alt="" />
                  <p className="icons-p"> {menu.rate}%</p>
                </div>
              </div>
              <p className="message">{menu.message}</p>
            </div>
          </div>
          <div className="grid-chart">
            <div className="sticky-container">
              <Cart />
            </div>
          </div>
          <div className="grid-sections">
            <div className="sticky-container">
              <Sections menu={menu} selectedCategory={selectedCategory} />
            </div>
          </div>
          <div className="grid-products">
            <form action="onSubmit" className="search-form">
              <label>
                <img src={magnifyingGlass} alt="" />
              </label>
              <input
                type="text"
                placeholder={`Search in ${menu.brandName}`}
                className="search-input"
              />
            </form>

            <Products
              selectedCategory={selectedCategory}
              menu={menu}
              loading={false}
              getCurrentSubMenu={getCurrentSubMenu}
            />
          </div>
        </div>
      </div>
      <div className="white-space"></div>
      <div className="transition-container">
        <footer className="transition" />
      </div>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  margin-top: -10rem;

  strong {
    padding-right: 0.5rem;
    font-family: "Gotham Office";
    color: black;
  }

  .grid-body {
    max-width: 156rem;

    display: grid;
    grid-template-columns: 0.5fr 4.25fr 1.5fr;
    grid-template-rows: auto 1fr;
    column-gap: 1rem;
    row-gap: 1.6rem;
  }

  .grid-title {
    max-width: 109rem;
    max-height: 40rem;
    grid-column: 1/3;
    grid-row: 1/2;
    box-shadow: 0 2px 20px rgb(0 0 0 / 10%);
    background: #fff;
    padding: 3rem 4rem;
    border-radius: 1rem;
  }

  .grid-chart {
    grid-row: 1/-1;
    grid-column: 3/4;
  }

  .sticky-container {
    position: sticky;
    top: 1rem;
  }

  .grid-sections {
    background-color: #fff;
    text-transform: uppercase;
    width: 17.5rem;
    height: auto;
    padding-bottom: 5rem;
  }

  .grid-products {
    background-color: #fff;
    padding: 1rem 0rem 1rem 0rem;
    height: auto;
    max-width: 120rem;
  }

  .grid-sections::-webkit-scrollbar {
    width: 0;
  }

  .grid-products::-webkit-scrollbar {
    width: 0;
  }

  .white-space {
    min-height: 2rem;
  }

  .transition-container {
    overflow: hidden;
    width: 100vw;
  }

  .transition {
    display: block;
    width: 110%;
    height: 12.5rem;
    background-color: #1d1d1d;
    border-top-left-radius: 50%;
    border-top-right-radius: 50%;
    margin: 0rem -5rem -5rem -5rem;
  }

  .brand-title {
    .message {
      font-size: 1.4rem;
      font-weight: 500;
      padding: 2rem 0rem;
      color: #1aa98f;
    }

    h1 {
      padding-top: 0.2rem;
    }
  }

  .icons {
    padding-top: 2rem;
    display: flex;
    gap: 2rem;
  }

  .del-icon-container {
    height: 2.6rem;
    p {
      letter-spacing: 0.1rem;
      font-size: 1.4rem;
      font-weight: 400;
    }
  }

  .del-icon {
    height: 2.8rem;
    padding-right: 0.8rem;
  }

  .search-form {
    width: 99%;
    background-color: #f5f5f5;
    padding: 1.25rem 1rem 1.2rem 1.8rem;
    border-radius: 10rem;
    label {
      padding-right: 1rem;
    }
  }

  .search-input {
    font-size: 1.6rem;
    background-color: #f5f5f5;
    border: none;
  }

  .search-input:focus {
    outline: none;
  }

  .sub-title {
    padding: 2rem 0 0 3rem;
    text-transform: uppercase;
  }

  h1 {
    font-size: 4.8rem;
    font-family: "Gotham", "Arial", sans-serif;
    font-weight: bold;
  }

  h2 {
    font-size: 3rem;
    letter-spacing: 0.05rem;
  }
`;
