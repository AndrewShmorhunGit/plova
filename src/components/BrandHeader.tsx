import useGeolocation from "react-hook-geolocation";
import { Link } from "react-router-dom";
import styled from "styled-components";
import { useAppSelector } from "../hooks/useAppDispatch";
import footerLogo1 from "../logos/footerLogo1.png";
import { IMenu } from "../modules/modules";
import { Path, UserData } from "./index";

export const BrandHeader = ({
  menu,
  loading,
}: {
  menu: IMenu;
  loading: boolean;
}) => {
  const location = useGeolocation();
  const address: string = `lat: ${location.latitude}, lng: ${location.longitude}`;
  const backgroundImage: string = menu.headerBackgroundImage;
  const { user } = useAppSelector((state) => state.user);
  return (
    <Wrapper>
      <header className="header">
        <div
          className="image-background"
          style={{
            backgroundImage: `linear-gradient(
        0deg,
        rgba(0, 0, 0, 0),
        rgba(0, 0, 0, 0.5)
      ),
      url(${backgroundImage})`,
          }}
        >
          <div className="container-products">
            <div className="header-top">
              <Link to="/">
                <img src={footerLogo1} alt="" className="header-logo" />
              </Link>

              <div
                className="header-address center"
                // onClick={() => setShowLocation(!showLocation)}
              >
                <div className="location">
                  <img
                    src="https://res.cloudinary.com/glovoapp/image/fetch//q_auto/https://glovoapp.com/images/svg/location2-white.svg"
                    alt="location marker"
                    className="header-location-icon"
                  />
                  <div className="header-text">
                    <span className="address-content-text">
                      {`${
                        address.length < 30
                          ? address
                          : address.slice(0, 30) + "..."
                      }`}
                    </span>
                  </div>
                  <img
                    src="https://res.cloudinary.com/glovoapp/image/fetch//q_auto/https://glovoapp.com/images/landing/dropdown-white.svg"
                    alt=""
                    role="presentation"
                    className="header__content__arrow"
                  />
                </div>
                <div className="center">
                  <UserData color={"white"} user={user} />
                </div>
              </div>
              {/* <div className="header-address">{address}</div> */}
            </div>
            <div className="path">
              <Path name={menu.brandName} />
            </div>
          </div>
        </div>
      </header>
    </Wrapper>
  );
};

const Wrapper = styled.header`
  .header {
    display: flex;
    flex-direction: column;
  }

  .image-background {
    padding-top: 2rem;
    background: linear-gradient(
      rgba(134, 142, 150, 0.5),
      rgba(241, 243, 245, 0.5)
    );

    background-size: cover;
    background-position: center;
    overflow: hide;
    height: 24.5rem;
  }

  .location {
    display: flex;
  }

  .header-address {
    padding-top: 0.9rem;
    color: white;
    font-size: 1.6rem;
    font-weight: 600;
  }

  .header-location-icon {
    height: 1.6rem;
    padding-right: 0.5rem;
    stretch: white;
  }

  .header__content__arrow {
    height: 2rem;
    color: white;
    padding-left: 0.15rem;
    margin-right: -0.8rem;
    margin-top: 0.15rem;
  }

  .menu-icon-user {
    padding: 0.1rem 1.6rem;
    cursor: pointer;
    margin-top: 0.1rem;
  }

  .menu-icon-orders {
    padding: 0.1rem 0.8rem;
    cursor: pointer;
    margin-top: 0.1rem;
  }

  .header-top {
    padding-top: 0.5rem;
    display: flex;
    justify-content: space-between;
  }

  .header-logo {
    width: 13rem;
  }

  .path {
    font-weight: 200;
    font-size: 1.6rem;
    color: white;
    padding-top: 1rem;
    text-align: start;
  }

  @media (max-width: 62.5em) {
    .location {
      display: none;
    }
  }
`;
