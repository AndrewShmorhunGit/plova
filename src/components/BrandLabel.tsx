import styled from "styled-components";

export const BrandLabel = () => {
  return (
    <Wrapper>
      <div className="container">
        <div className="brand-title">
          <h1>Brand name</h1>
          <div className="icons">
            <p>
              <strong>🕔</strong>
              {`delivery time`}'
            </p>
            <p>
              <strong>🚀</strong>
              {`deliv price`} $
            </p>
            <p>
              <strong>👍</strong>
              {`rate`} %
            </p>
          </div>
        </div>
      </div>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  .brand-title {
    font-size: 4rem;
    background: #fff;
    border: none;
    border-radius: 1rem;
    padding: 6rem 6rem;
  }

  .icons {
    padding-top: 2rem;
    display: flex;
    gap: 2rem;
    font-size: 2rem;
  }

  strong {
    padding-right: 0.5rem;
  }
`;
