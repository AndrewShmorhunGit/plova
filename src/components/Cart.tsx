import { useLocation } from "react-router-dom";
import styled from "styled-components";
import { useAppSelector } from "../hooks/redux";
import { getCurrentCard } from "../store/actions/cartActions";
import { getSlugFromLocation } from "../units/functions";
import { CartUnit } from "./index";

export const Cart = () => {
  const location = useLocation();
  const slug = getSlugFromLocation(location);
  const { carts } = useAppSelector((state) => state.carts);

  const currentCart = getCurrentCard(slug, carts);

  return (
    <Wrapper>
      <div className="chart">
        <h2 className="empty-chart-title center">
          {currentCart === null ? "Your plova" : "Your order"}
        </h2>
        {currentCart === null ? (
          <div className="cart-inits-container">
            <div>
              <img
                className="empty-chart-image"
                src="https://res.cloudinary.com/glovoapp/image/fetch//q_auto/https://glovoapp.com/images/svg/astronaut-grey-scale.svg"
                alt="spaceman around the food"
              />
            </div>
            <div>
              <p className="empty-chart-paragraph center">
                You've not added any products yet. When you do, you'll see them
                here!
              </p>
            </div>
          </div>
        ) : (
          <div className="cart-units-container">
            {currentCart &&
              currentCart.order.map((singleOrder, index) => {
                return (
                  <CartUnit key={index} singleOrder={singleOrder} slug={slug} />
                );
              })}
          </div>
        )}
      </div>
      <div className="chart-bottom center">
        <div className="center">
          <img
            className="chart-bottom-image"
            src="https://res.cloudinary.com/glovoapp/image/fetch//q_auto/https://glovoapp.com/images/svg/feedback/info.svg"
            alt=""
          />
          <p>
            Reach <span>5,00 $</span> to save <span>1,00 $</span> in fees!
          </p>
        </div>
        {currentCart !== null && (
          <div className="center">
            <button className="order-btn btn">
              Make an Order (`1`) <br /> for (totalPrice)
            </button>
          </div>
        )}
      </div>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  .margin-bottom {
    margin-bottom: -10rem;
  }

  .chart {
    display: flex;
    flex-direction: column;
    gap: 3.6rem;
    padding: 4.4rem 2rem 4rem 2rem;
    min-height: 30rem;
  }

  .empty-chart-paragraph {
    font-size: 1.6rem;
    line-height: 1.5;
    text-align: center;
    font-weight: 400;
    color: #4d4d4d;
  }

  .empty-chart-image {
    display: flex;
    height: 18rem;
    width: 100%;
  }

  .cart-units-container {
    overflow-y: scroll;
    max-height: 30rem;
    min-height: 20rem;

    ::-webkit-scrollbar {
      width: 1rem;
    }

    ::-webkit-scrollbar-track {
      background: #e9f8f5;
      border-radius: 1rem;
    }

    ::-webkit-scrollbar-thumb {
      background: #00a082;
      opacity: 50%;
      border-radius: 1rem;
    }
  }

  .chart-bottom {
    position: sticky;
    top: 2rem;
    flex-direction: column;
    gap: 2rem;
    padding: 1rem 0rem;
    font-size: 1.2rem;
    font-weight: 400;
    border-top: 4px solid #e9f8f5;
    color: #4d4d4d;
  }

  .chart-bottom-image {
    margin-bottom: 0.1rem;
    max-height: 1.4rem;
    padding-right: 0.4rem;
  }

  .order-btn {
    height: auto;
    max-width: auto;
    padding: 2rem 4rem;
    font-size: 1.8rem;
    letter-spacing: 0rem;
    border-radius: 100rem;
  }
`;
