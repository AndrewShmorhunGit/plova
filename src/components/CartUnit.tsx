import styled from "styled-components";
import { useAppDispatch } from "../hooks/redux";
import { Order } from "../modules/modules";
import { cartSlice } from "../store/slices/cartSlice";
import { showDollarPrice } from "../units/functions";
import plus from "../images/menu/plusNew.svg";
import minus from "../images/menu/minusNew.svg";

export const CartUnit = ({
  singleOrder,
  slug,
}: {
  singleOrder: Order;
  slug: string;
}) => {
  const dispatch = useAppDispatch();
  const increase = () => {
    dispatch(
      cartSlice.actions.changeCartAmount({
        slug,
        name: singleOrder.name,
        operation: "inc",
      })
    );
  };

  const decrease = () => {
    dispatch(
      cartSlice.actions.changeCartAmount({
        slug,
        name: singleOrder.name,
        operation: "dec",
      })
    );
  };
  return (
    <Wrapper>
      <div className="center">
        <div className="orders-container">
          <div>
            <p className="amount"> x{singleOrder.amount} </p>
          </div>
          <div>
            <p className="name">{singleOrder.name}</p>
          </div>
          <div>
            <p className="total-price">
              {showDollarPrice(singleOrder.price * singleOrder.amount)}$
            </p>
          </div>
          <div className="buttons">
            <img
              onClick={() => decrease()}
              className="dec-btn"
              src={minus}
              alt=""
            />

            {/* <button className="edit-btn">Edit</button> */}

            <img
              onClick={() => increase()}
              className="inc-btn"
              src={plus}
              alt=""
            />
          </div>
        </div>
      </div>
    </Wrapper>
  );
};
const Wrapper = styled.main`
  .orders-container {
    max-width: 100%;
    display: grid;
    grid-template-columns: 0.5fr 4fr 1fr;
    column-gap: 2rem;
    font-weight: 500;
    padding: 1rem 1rem 0.5rem 1rem;
    border-bottom: 1px solid #e9f8f5;
    margin-right: 0.5rem;
  }

  .amount {
    font-size: 2rem;
    font-weight: 600;
  }

  .name {
    font-size: 1.8rem;
    font-weight: 400;
    padding-bottom: 1rem;
  }

  .total-price {
    font-size: 1.8rem;
    font-weight: 600;
  }

  .dec-btn,
  .inc-btn {
    cursor: pointer;
  }

  .buttons {
    grid-column: 1/-1;
    // grid-row: 1/;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }

  .edit-btn {
    color: #00a082;
    font-weight: 600;
    background: none;
    border: none;
    cursor: pointer;
    display: inline-block;
    width: 5rem;
    padding: 0;
  }
`;
