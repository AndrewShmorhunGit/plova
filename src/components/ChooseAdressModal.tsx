const ChooseAddressModal = () => {
  return (
    <div>
      <h1>Cho0os Address</h1>
    </div>
  );
};

export default ChooseAddressModal;
