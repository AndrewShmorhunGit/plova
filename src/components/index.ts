import { Cart } from "./Cart";
import { Jumbotron } from "./Jumbotron";
import { FoodNearGallery } from "./FoodNearGallery";
import { Footer } from "./Footer";
import { BrandHeader } from "./BrandHeader";
import { LocalSearch } from "./LocalSearch";
import { MainHeader } from "./MainHeader";
import { Path } from "./Path";
import { PopularNear } from "./PopularNear";
import { Products } from "./Products";
import { SearchInput } from "./SearchInput";
import { SingleProduct } from "./SingleProduct";
import { StartOrderWindow } from "./StartOrderWindow";
import { UserComponent } from "./UserComponent";
import { ChooseLocation } from "./ChooseLocation";
import { Registration } from "./Registration";
import { Partnership } from "./Partnership";
import { LocationInput } from "./LocationInput";
import { SearchLocationModal } from "./SearchLocationModal";
import { MapSearch } from "./MapSearch";
import { Sections } from "./Sections";
import { InputHeader } from "./InputHeader";
import { BrandGrid } from "./BrandGrid";
import { GalleryLoader } from "./GalleryLoader";
import { Map } from "./Map";
import { LoadingHeader } from "./LoadingHeader";
import { LoadingBrandGrid } from "./LoadingBrandGrid";
import { LoadingProducts } from "./LoadingProducts";
import { CartUnit } from "./CartUnit";
import { ExitFromOrderModal } from "./ExitFromOrderModal";
import { AllergyModal } from "./AllergyModal";
export {
  AllergyModal,
  ExitFromOrderModal,
  CartUnit,
  LoadingProducts,
  LoadingBrandGrid,
  LoadingHeader,
  Map,
  GalleryLoader,
  BrandGrid,
  InputHeader,
  Sections,
  Footer,
  Cart,
  Jumbotron,
  FoodNearGallery,
  BrandHeader,
  LocalSearch,
  MainHeader,
  Path,
  PopularNear,
  Products,
  SearchInput,
  SingleProduct,
  StartOrderWindow,
  UserComponent,
  ChooseLocation,
  Registration,
  Partnership,
  LocationInput,
  SearchLocationModal,
  MapSearch,
};
