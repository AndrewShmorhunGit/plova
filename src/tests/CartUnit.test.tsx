import { CartUnit } from "../components/CartUnit";
import { renderWithProviders } from "./utils/test-utils";
import * as reduxHooks from "react-redux";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
// import * as actions from "../store/slices/cartSlice";
// import { useAppDispatch } from "../hooks/useAppDispatch";
// import { testUseAppSelector } from "./utils/test-app-selector";
// import { useAppSelector } from "../hooks/useAppDispatch";

const fakeOrder = { amount: 5, name: "Spaghetti", price: 6.2 };
jest.mock("react-redux");
const mockDispatch = jest.spyOn(reduxHooks, "useDispatch");

describe("CartUnit", () => {
  beforeEach(() => {
    // useAppSelector.mockImplementation(testUseAppSelector);
    // useAppDispatch.mockImplementation(() => jest.fn);
  });
  afterEach(() => jest.clearAllMocks);

  it("create snapshot", () => {
    mockDispatch.mockReturnValue(jest.fn());
    const component = (
      <CartUnit singleOrder={fakeOrder} slug={"anyRestaurant"} />
    );
    renderWithProviders(component, {
      // preloadedState: {
      //   carts: {
      //     carts: {
      //       anyRestaurant: {
      //         order: [fakeOrder],
      //       },
      //     },
      //   },
      // },
    });

    expect(component).toMatchSnapshot();
  });

  it("component display all parameters", () => {
    // renderWithProviders(
    //   <CartUnit singleOrder={fakeOrder} slug={"anyRestaurant"} />
    // );

    render(<CartUnit singleOrder={fakeOrder} slug={"anyRestaurant"} />);

    const price = screen.getByRole("price");
    const position = screen.getByRole("position");
    const amount = screen.getByRole("amount");
    expect(price).toHaveTextContent("31,00$");
    expect(position).toHaveTextContent("Spaghetti");
    expect(amount).toHaveTextContent("x5");
  });

  it("should call dispatch", () => {
    render(<CartUnit singleOrder={fakeOrder} slug={"anyRestaurant"} />);

    const increment = screen.getByRole("increase", { name: /plus in circle/i });

    userEvent.click(increment);

    expect(mockDispatch).toHaveBeenCalled();
  });

  it("check component increment and decrement amount in position", () => {
    const component = (
      <CartUnit singleOrder={fakeOrder} slug={"anyRestaurant"} />
    );
    const result = renderWithProviders(component, {
      preloadedState: {
        carts: {
          carts: {
            anyRestaurant: {
              order: [fakeOrder],
            },
          },
        },
      },
    });
    const dispatch = jest.fn;
    mockDispatch.mockReturnValue(dispatch);

    // const mackedIncreaseValue = jest.spyOn(actions, "changeCartAmount");

    result.debug();

    const increment = result.getByRole("increase");
    const decrement = result.getByRole("decrease", {
      name: /minus in circle/i,
    });
    const amount = result.getByRole("amount");
    userEvent.click(increment);
    expect(amount).toHaveTextContent("x6");
    userEvent.click(decrement);
    expect(amount).toHaveTextContent("x5");
  });
});
