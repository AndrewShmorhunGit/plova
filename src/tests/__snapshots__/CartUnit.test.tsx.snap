// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartUnit check component increment and decrement amount in position 1`] = `<DocumentFragment />`;

exports[`CartUnit create snapshot 1`] = `
<CartUnit
  singleOrder={
    Object {
      "amount": 5,
      "name": "Spaghetti",
      "price": 6.2,
    }
  }
  slug="anyRestaurant"
/>
`;
